common/src/stack/ansible/plugins/module_utils/stacki.py

```diff
@@ -0,0 +1,50 @@
+# @copyright@
+# Copyright (c) 2006 - 2020 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import json
+import subprocess
+
+
+class StackCommandError(Exception):
+    """Exception raised for errors running a stack command."""
+
+    def __init__(self, message):
+        self.message = message
+
+
+def run_stack_command(command, args=None):
+	"""Runs a stack command, returning StackCommandError if there is an error."""
+
+	# Create our command to run
+	command_args = ["/opt/stack/bin/stack"]
```

Comment:
This effectively ties the ansible modules to running on the frontend, no?

Reply:
Correct, which I decided was appropriate for the first version/MVP. Since our Ansible playbooks will be running on the frontend anyways, and the Ansible modules only installed on the frontend during the normal workflow, it should satisfy SNAPI's needs.

In a version 2, especially if we change our Webservice to be JWT based, it might make sense to do the work to have this function use the webservice and deal with authentication and whatnot. If we switch to using JWT, we would only need to supply an API endpoint and the JWT refresh token in any playbook.

But, for the time being is simpler to just assume the modules are running on the frontend.

Reply:
Also, maybe we already talked about this and I forgot, but is there a reason to shell out to `stack` rather than `import stack` and handle it all via Python?

Reply:
Part of it for simplicity reasons, I'm not sure how loading all of Stacki's python stuff into Ansible's python process would go. Also, I'm concerned about the Stacki DB cache layer, we saw lots of bugs with Apache not shelling out to run Stacki commands.

What are the reasons to *not* shell out, besides the overhead of spinning up a process?